qiskit_dynamics/dispatch/__init__.py

```diff
@@ -10,10 +12,189 @@
+
+* :attr`~Array.data` which returns the wrapped array object
+* :attr`~Array.backend` which returns the backend string of the wrapped array.
+
+All other attributes and methods of the wrapped array are accessable
+through this class, but with any array return types wrapped into
+:class:`Array` objects. For example
+
+.. jupyter-execute::
+
+    # Call numpy.ndarray method
+    a.reshape((2, 5))
+
+Array Initialization
+--------------------
+
+An `Array` object can be initialized as `Array(data)` from any ``data`` that is
+
+1. An ``array`` object of one of the supported backends.
+2. An object that can be used to initialize the backend array.
+3. An object of a class that defines a `__qiskit_array__` method with the
+   following signature
+
+    .. code-block:: python
+
+        def __qiskit_array__(self,
+                             dtype: Optional[any] = None,
+                             backend: Optional[str]=None) -> Array:
+            # conversion from class to Array
+
+The :meth:`Array.__init__` method has optional kwargs
+
+* ``dtype`` (``any``): Equivalent to the ``numpy.array`` ``dtype`` kwarg.
+  For other array backends the specified dtype must be supported on the
+  array backend. If not specified the dytpe will be inferred from the
+  input data.
+* ``order`` (``str``): Equivalent to the ``numpy.array`` ``order`` kwarg.
+  For other array backendsthe specified order value must be supported on
+  the array backend.
+* ``backend`` (``str``): The array backend to use. If not specified this
+  will be inferred from the input data type if it is a backend array
+  instance, otherwise it will be the :func:`default_backend`.
+
+
+Using Arrays with other Libraries
+=================================
+
+The :class`Array` class is intended to be used with array functions in
+Python libraries that work with any of the supported Array backends.
+
+Wrapping Functions
+------------------
+
+Functions from other libraries can be wrapped to work with Array objects using
+the :func:`wrap` function as
+
+.. code-block:: python
+
+    from qiskit_dynamics.dispatch import Array, wrap
+
+    # Some 3rd-party library
+    import library
+
+    # Wrap library function f
+    f = dispatch.wrap(library.f)
+
+    # Apply f to an Array
+    a = Array([0, 1, 2, 4])
+    b = f(a)
+
+The wrapped function will automatically unwrapping any Array args and kwargs
+for calling the wrapped function. If the result of the function is a backend array
+or an instance of a class that defines a `__qiskit_array__` method the wrapped
+function will return this as an Array object. Similarly if the result is a tuple
+this conversion will be applied to each element of the tuple where appropriate.
+
+Wrapping Decorators
+-------------------
+
+The :func:`wrap` function can also be used to wrap function decorators by setting
+the ``decorator=True`` kwarg.
+
+For example to wrap autograd and jit decorators from the Kax library to work with
```

Comment:
```suggestion
For example to wrap autograd and jit decorators from the JAX library to work with
```

We should try to consistently refer to JAX using all caps, as that is how the JAX team writes it in their own docs.